### ISSUE TITLE
Sample Field Bugs

### ISSUE THREAD

**Issue by aaronstephenson** [USGS-WiM/mercurymobile#41]
1.) I entered the value "1234" into the Time field, intending "12:34 PM", but after saving, then re-opening the sample, the Time value is "1:23:00". I think this is a bug I introduced months ago when parsing the string value from that field on save.

![sample_bug_time](https://user-images.githubusercontent.com/8713712/37123358-8cde5a0a-2229-11e8-8961-e4819554e48a.PNG)

2.) When opening a sample, more often than not the Project Name field is empty, even when the Project ID field is populated. When the Project ID is selected (or pre-selected, in the case of situations where there is only one project in the Project DB), the Project Name should also be automatically selected to match the ID (and also written to the sample record on save).

![sample_bug_project_name](https://user-images.githubusercontent.com/8713712/37123443-d4d1d882-2229-11e8-95d7-ff8a8fcb0b95.PNG)

Additionally, I'm seeing fields whose data types should be Number containing String values. 

![sample_fields_bugs](https://user-images.githubusercontent.com/8713712/37123781-0dc532f0-222b-11e8-910c-50dafa0b7f5b.PNG)

**Comment by ehaberman-usgs** [comment]
Fixed.

Converted the time field to a time selector -- format 00:00:00 -- should we use 24 H here? Currently 1 - 12

**Comment by aaronstephenson** [comment]
@ehaberman-usgs re: the time field, the lab staff specifically didn't want a time selector. I had used that originally, but they only want a plain text field to manually type in the time value.

So the outstanding problem is that the value getting entered is not the value getting saved (in my case, I typed 1234 but the value that was saved was 1:23:00), and the value displayed after a save still has the colons, when it should just be a four digit string.
